### PR TITLE
Make Carousel component headless.

### LINKED ISSUE #1857: Make the `@yamada-ui/carousel` component headless
script: make-component-headless
package: @yamada-ui/carousel

### Description
Currently, the `@yamada-ui/carousel` component has minimal styles set within the component's source code.
We want to make this style transferable to the theme and make the component headless, enabling more flexible UI customization.

### Problem Statement/Justification
By making the `@yamada-ui/carousel` component headless, we can achieve more flexible styling tailored to the project.

### Proposed Solution or API
The styles within the component are mostly defined by `css: CSSUIObject` and set to a prop named `__css`. We will transfer these styles to the `packages/theme/src/components`.

The code below is just an example and not the actual code.

`packages/components/input/src/input-group.tsx`
```diff
const css: CSSUIObject = {
- width: "100%",
- display: "flex",
- position: "relative",
  ...styles.container,
}
```

`packages/theme/src/components/input.ts`
```diff
baseStyle: {
  container: {
+   width: "100%",
+   display: "flex",
+   position: "relative",
  },
  field: {
    width: "100%",
    minWidth: 0,
    outline: 0,
    position: "relative",
    appearance: "none",
    transitionProperty: "common",
    transitionDuration: "normal",
    _disabled: {
      opacity: 0.4,
      cursor: "not-allowed",
    },
  },
  element: {
    color: ["blackAlpha.600", "whiteAlpha.700"],
  },
},
```

> [!TIP]
>
> In cases where the styles within the component are dynamically changed, there may be instances where it is not possible to transfer the styles to the `theme`.
> In such cases, it needs to be discussed in this issue. Please feel free to send comments to this issue.

### Additional Information
The `@yamada-ui/carousel` component exists in `packages/components/carousel/src`. You need to check all the components in this folder and if they contain styles internally, you need to modify them.

This change could have a significant impact on projects using Yamada UI, so we will work on it within the scope of `v2.0`.
Please create the branch for working on this issue from the remote branch `v2.0`, and direct your pull requests to the `v2.0` branch as well.

Please select `major` for the changeset and create a pull request.

### PR DESCRIPTION
**Pull request by balckowl**
Closes #1857

## Description

Make Carousel component headless.

## Is this a breaking change (Yes/No):

No.

## Additional Information

I left the CSSUIObject on line 315 of packages/components/carousel/src/carousel.tsx as is because I could not find the migration destination in packages/theme/src/components/carousel.ts.

### CODE REVIEW

File: packages/components/carousel/src/carousel-indicators.tsx
Comment (108yen) [inline packages/components/carousel/src/carousel-indicators.tsx:1]: Line 75 is affected.
Is this not a problem?

File: packages/components/carousel/src/carousel-slide.tsx
Comment (108yen) [inline packages/components/carousel/src/carousel-slide.tsx:1]: Unnecessary change.

File: packages/components/carousel/src/carousel.tsx
Comment (108yen) [inline packages/components/carousel/src/carousel.tsx:1]: You missed one place.

File: packages/theme/src/components/carousel.ts [hunk 1]
```diff
@@ -5,9 +5,16 @@ export const Carousel: ComponentMultiStyle = {
...
     },
-    inner: {},
-    slide: {},
-    control: {},
+    inner: {
+      display: "flex",
+    },
+    slide: {
+      position: "relative",
+    },
```
Comment (108yen) [inline packages/theme/src/components/carousel.ts:13]: Unnecessary change.

**Review by 108yen — CHANGES_REQUESTED** [review]

File: packages/theme/src/components/carousel.ts (hunk 1 above)
Comment (balckowl) [reply packages/theme/src/components/carousel.ts:13]: Will it be okay if I revert this part?

Comment (108yen) [reply packages/theme/src/components/carousel.ts:13]: Yes.

Comment (balckowl) [reply packages/theme/src/components/carousel.ts]: I have fixed it. Please confirm.

**Review by fffk10 — COMMENTED** [review]
I've confirmed.
I don't think there are any problems other than the points that have already been pointed out.

**Comment by hirotomoyamada** [comment]
@108yen 

Please re-reviews.

**Comment by hirotomoyamada** [comment]
@108yen @fffk10 

Please approval.

File: packages/components/carousel/src/carousel.tsx
Comment (108yen) [reply packages/components/carousel/src/carousel.tsx:1]: @balckowl
You missed the location of `styles.container`.

**Review by fffk10 — DISMISSED** [review]

**Review by 108yen — APPROVED** [review]

**Review by hirotomoyamada — APPROVED** [review]

**Comment by hirotomoyamada** [comment]
Thanks!

**Pull request merged**